lib/fileset/default.nix

```diff
@@ -632,55 +758,72 @@ in {
       (elemAt filesets 0)
       (elemAt filesets 1);
 
-  /*
+  /**
     Filter a file set to only contain files matching some predicate.
 
-    Type:
-      fileFilter ::
-        ({
-          name :: String,
-          type :: String,
-          hasExt :: String -> Bool,
-          ...
-        } -> Bool)
-        -> Path
-        -> FileSet
-
-    Example:
-      # Include all regular `default.nix` files in the current directory
-      fileFilter (file: file.name == "default.nix") ./.
-
-      # Include all non-Nix files from the current directory
-      fileFilter (file: ! file.hasExt "nix") ./.
-
-      # Include all files that start with a "." in the current directory
-      fileFilter (file: hasPrefix "." file.name) ./.
-
-      # Include all regular files (not symlinks or others) in the current directory
-      fileFilter (file: file.type == "regular") ./.
-  */
-  fileFilter =
-    /*
-      The predicate function to call on all files contained in given file set.
-      A file is included in the resulting file set if this function returns true for it.
 
-      This function is called with an attribute set containing these attributes:
+    # Inputs
+
+    `predicate`
+
+    : The predicate function to call on all files contained in given file set.
+    A file is included in the resulting file set if this function returns true for it.
+
+    This function is called with an attribute set containing these attributes:
 
-      - `name` (String): The name of the file
+    - `name` (String): The name of the file
 
-      - `type` (String, one of `"regular"`, `"symlink"` or `"unknown"`): The type of the file.
-        This matches result of calling [`builtins.readFileType`](https://nixos.org/manual/nix/stable/language/builtins.html#builtins-readFileType) on the file's path.
+    - `type` (String, one of `"regular"`, `"symlink"` or `"unknown"`): The type of the file.
+      This matches result of calling [`builtins.readFileType`](https://nixos.org/manual/nix/stable/language/builtins.html#builtins-readFileType) on the file's path.
 
-      - `hasExt` (String -> Bool): Whether the file has a certain file extension.
-        `hasExt ext` is true only if `hasSuffix ".${ext}" name`.
+    - `hasExt` (String -> Bool): Whether the file has a certain file extension.
+      `hasExt ext` is true only if `hasSuffix ".${ext}" name`.
 
-        This also means that e.g. for a file with name `.gitignore`,
-        `hasExt "gitignore"` is true.
+      This also means that e.g. for a file with name `.gitignore`,
+      `hasExt "gitignore"` is true.
 
-      Other attributes may be added in the future.
-    */
+    Other attributes may be added in the future.
```

Comment:
Also missing extra indentation on many of the lines here, they should all be part of the content for `predicate`.